include/zos-filesystem.h

```diff
@@ -0,0 +1,9 @@
+#ifndef ZOS_FILESYSTEM_H
+#define ZOS_FILESYSTEM_H
+
+#include <string.h>
```

Comment:
We may want to guard this with #if defined(__cplusplus), in case it gets included in a C project indirectly